Fix various issues discovered by coverity

kgaillot opened the pull request:
Most of these are insignificant or false positives, but they do tighten up some edge cases.

beekhof commented on crmd/messages.c:
```diff
@@ -805,8 +805,8 @@ handle_request(xmlNode * stored_msg, enum crmd_fsa_cause cause)
-        if(msg) {
-            relay_message(msg, TRUE);
+        if (msg && (relay_message(msg, TRUE) == FALSE)) {
+            crm_notice("Unable to reply to CRM ping");
```
`relay_message()` already does logging. Maybe `debug` instead.

kgaillot commented:
Amended commits based on prior discussion: rebased on current master; removed commit updating bundled md5 code, due to license change; removed redundant log message; restored CRM_CHECKs, at later point in code; fixed client->id bug.
